src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -439,6 +441,52 @@ public async Task RemoveFromClusterAsync(string nodeTag)
             await _engine.RemoveFromClusterAsync(nodeTag).WithCancellation(_shutdownNotification.Token);
         }
 
+        public async Task RequestSnapshot()
+        {
+            var topology = GetClusterTopology();
+
+            if (topology.AllNodes.Count == 1)
+                throw new InvalidOperationException("Can't force snapshot, since I'm the only node in the cluster.");
+
+            var leaderTag = _engine.LeaderTag;
+            var leaderUrl = topology.GetUrlFromTag(leaderTag);
+
+            if (leaderTag == null || leaderUrl == null)
+                throw new NoLeaderException("Can't force snapshot, no leader found.");
+
+            var isMember = topology.Members.ContainsKey(NodeTag);
+            using (var requestExecutor = CreateNewClusterRequestExecutor(leaderUrl))
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            {
+                if (isMember)
+                {
+                    if (IsLeader())
+                    {
+                        Engine.CurrentLeader.StepDown();
+                    }
+
+                    await WaitForState(RachisState.Follower, ServerShutdown);
+
+                    // demote to non-voter, so we will not participate in the quorum.
+                    var demote = new DemoteClusterNodeCommand(NodeTag);
+                    await requestExecutor.ExecuteAsync(demote, ctx, token: ServerShutdown);
+                }
+
+                using (var tx = ctx.OpenWriteTransaction())
```

Comment:
No, not to all of this.
It is _really_ complex and have too many steps in the middle.
Instead, use:
* cannot call request snapshot on the leader (user can demote manually).

And that solves the whole issue for you.